src/Microsoft.Diagnostics.ExtensionCommands/Host/RuntimeInfo.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Text;
+using Microsoft.Diagnostics.DebugServices;
+using Microsoft.FileFormats;
+
+namespace Microsoft.Diagnostics.Runtime
+{
+    public class RuntimeInfo : TStruct
+    {
+        public const string RUNTIME_INFO_SYMBOL = "DotNetRuntimeInfo";
+        public const string RUNTIME_INFO_SIGNATURE = "DotNetRuntimeInfo";
+        public const int RUNTIME_INFO_RUNTIME_VERSION = 2;
+        public const int RUNTIME_INFO_LATEST = 2;
+
+        [ArraySize(18)]
+        public readonly byte[] RawSignature;
+        public readonly int Version;
+        [ArraySize(24)]
+        public readonly byte[] RawRuntimeModuleIndex;
+        [ArraySize(24)]
+        public readonly byte[] RawDacModuleIndex;
+        [ArraySize(24)]
+        public readonly byte[] RawDbiModuleIndex;
+        [ArraySize(4)]
+        public readonly int[] RawRuntimeVersion;                // major, minor, build, revision - added in version RUNTIME_INFO_RUNTIME_VERSION
+
+        public static unsafe bool TryRead(IServiceProvider services, ulong address, out RuntimeInfo info)
+        {
+            info = default;
+
+            Reader reader = services.GetService<Reader>();
+            if (reader is null)
+            {
+                return false;
+            }
+
+            try
+            {
+                info = reader.Read<RuntimeInfo>(address);
+            }
+            catch (Exception ex) when (ex is InvalidVirtualAddressException or BadInputFormatException)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        public string Signature => Encoding.ASCII.GetString(RawSignature.Take(RUNTIME_INFO_SIGNATURE.Length).ToArray());
```

Comment:
What should be span overloads?  The Signature property result? RawSignature (FileFormats probably doesn't support that).  There is no encoding GetString function that takes a span.

Does it make that much difference since this is used by some commands and only allocates anything when the commands are run? Which is why I didn't cache the string either.

Reply:
> There is no encoding GetString function that takes a span

https://learn.microsoft.com/en-us/dotnet/api/system.text.encoding.getstring?view=net-8.0#system-text-encoding-getstring(system-readonlyspan((system-byte)))

essentially `Encoding.ASCII.GetString(RawSignature.AsSpan(0, RUNTIME_INFO_SIGNATURE.Length))`, but at the same time I realized that was ns2.1 so disregard this particular one. I do think we need to start being more conscious over allocations if we are going to rely on SOS for heavy lifting of so many plans.